File: src/main/java/com/atsuishio/superbwarfare/client/renderer/entity/SteelCoilRenderer.java
package com.atsuishio.superbwarfare.client.renderer.entity;

import com.atsuishio.superbwarfare.client.model.entity.SteelCoilModel;
import com.atsuishio.superbwarfare.entity.projectile.SteelCoilEntity;
import com.mojang.blaze3d.vertex.PoseStack;
import com.mojang.blaze3d.vertex.VertexConsumer;
import com.mojang.math.Axis;
import net.minecraft.client.renderer.MultiBufferSource;
import net.minecraft.client.renderer.RenderType;
import net.minecraft.client.renderer.entity.EntityRendererProvider;
import net.minecraft.resources.ResourceLocation;
import org.jetbrains.annotations.Nullable;
import software.bernie.geckolib.cache.object.BakedGeoModel;
import software.bernie.geckolib.renderer.GeoEntityRenderer;

public class SteelCoilRenderer extends GeoEntityRenderer<SteelCoilEntity> {

    public SteelCoilRenderer(EntityRendererProvider.Context renderManager) {
        super(renderManager, new SteelCoilModel());
        this.shadowRadius = 0f;
    }

    @Override
    public RenderType getRenderType(SteelCoilEntity animatable, ResourceLocation texture, MultiBufferSource bufferSource, float partialTick) {
        return RenderType.entityTranslucent(getTextureLocation(animatable));
    }

    @Override
    public void actuallyRender(PoseStack poseStack, SteelCoilEntity animatable, BakedGeoModel model, @Nullable RenderType renderType, MultiBufferSource bufferSource, @Nullable VertexConsumer buffer, boolean isReRender, float partialTick, int packedLight, int packedOverlay, int colour) {
        poseStack.pushPose();

        if (animatable.getDeltaMovement().lengthSqr() > 0) {
            poseStack.mulPose(Axis.YP.rotationDegrees(animatable.yRotO));
//            poseStack.mulPose(Axis.XP.rotationDegrees(Mth.lerp(partialTick, animatable.xRotO, animatable.getXRot())));
        }
        super.actuallyRender(poseStack, animatable, model, renderType, bufferSource, buffer, isReRender, partialTick, packedLight, packedOverlay, colour);

        poseStack.popPose();
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/client/renderer/entity/SteelCoilRenderer.kt
package com.atsuishio.superbwarfare.client.renderer.entity

import com.atsuishio.superbwarfare.client.model.entity.SteelCoilModel
import com.atsuishio.superbwarfare.entity.living.SteelCoilEntity
import net.minecraft.client.renderer.MultiBufferSource
import net.minecraft.client.renderer.RenderType
import net.minecraft.client.renderer.entity.EntityRendererProvider
import net.minecraft.client.renderer.texture.OverlayTexture
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.renderer.GeoEntityRenderer

class SteelCoilRenderer(renderManager: EntityRendererProvider.Context) : GeoEntityRenderer<SteelCoilEntity>(
    renderManager,
    SteelCoilModel()
) {
    init {
        this.scaleWidth = 2f
        this.scaleHeight = 2f
    }

    override fun getRenderType(
        animatable: SteelCoilEntity,
        texture: ResourceLocation?,
        bufferSource: MultiBufferSource?,
        partialTick: Float
    ): RenderType = RenderType.entityTranslucent(getTextureLocation(animatable))

    override fun getDeathMaxRotation(entityLivingBaseIn: SteelCoilEntity): Float {
        return 0f
    }

    override fun shouldShowName(animatable: SteelCoilEntity): Boolean {
        return animatable.hasCustomName()
    }

    @Suppress("removal")
    override fun getPackedOverlay(animatable: SteelCoilEntity?, u: Float): Int {
        return OverlayTexture.pack(OverlayTexture.u(u), OverlayTexture.v(false))
    }
}